rest-api-spec/src/main/resources/rest-api-spec/api/fleet.search.json

```diff
@@ -0,0 +1,50 @@
+{
+  "fleet.search":{
+    "documentation":{
+      "url": null,
+      "description": "Search API where the search will only be executed after specified checkpoints are available due to a refresh. This API is designed for internal use by the fleet server project."
+    },
+    "stability":"experimental",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "application/json"],
+      "content_type": ["application/json"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/{index}/_fleet/_search",
+          "methods":[
+            "GET",
+            "POST"
+          ],
+          "parts":{
+            "index":{
+              "type":"string",
+              "description":"The name of the index."
+            }
+          }
+        }
+      ]
+    },
+    "params":{
+      "wait_for_checkpoints":{
+        "type":"list",
+        "description":"Comma separated list of checkpoints, one per shard",
+        "default":""
+      },
+      "timeout":{
```

Comment:
`timeout` is a bit overloaded, maybe `wait_for_checkpoint_timeout` to avoid ambiguity ?